plaso/parsers/gdrive_synclog.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+"""Parser for Google Drive Sync log files."""
+
+from __future__ import unicode_literals
+
+import logging
+
+import pyparsing
+
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import errors
+from plaso.lib import definitions
+from plaso.parsers import manager
+from plaso.parsers import text_parser
+
+
+class GDriveSyncLogEventData(events.EventData):
+  """GDrive Sync log event data.
+
+  Attributes:
+    time (str): time of event (string, time with offset and timezone of user).
+    log_level (str): logging level of event (e.g., DEBUG, WARN, INFO, ERR).
+    pid (int): process ID of process which logged event.
+    thread (str): ID:name of thread which logged event (colon-separated).
+    source_code (str): filename:line_number of source file which logged event.
+    message (str): log message.
+  """
+
+  DATA_TYPE = 'gdrive_sync:log:line'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(GDriveSyncLogEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.time = None
+    self.log_level = None
+    self.pid = None
+    self.thread = None
+    self.source_code = None
+    self.message = None
+
+
+class GDriveSyncLogParser(text_parser.PyparsingMultiLineTextParser):
+  """Parses events from GDrive Sync log files."""
+
+  NAME = 'gdrive_synclog'
+
+  DESCRIPTION = 'Parser for GDrive Sync log files.'
+
+  # TODO: confirm this is ascii on OS X, Linux?
+  _ENCODING = 'ascii'
+
+  # Common structures.
+  _HYPHEN = text_parser.PyparsingConstants.HYPHEN
+
+  _FOUR_DIGITS = text_parser.PyparsingConstants.FOUR_DIGITS
+  _THREE_DIGITS = text_parser.PyparsingConstants.THREE_DIGITS
+  _TWO_DIGITS = text_parser.PyparsingConstants.TWO_DIGITS
+
+  # Ignore commas and spaces.
+  IGNORE_FIELD = pyparsing.CharsNotIn(',').suppress()
+
+  _GDS_DATE_TIME = (
+      _FOUR_DIGITS.setResultsName('year') + _HYPHEN +
+      _TWO_DIGITS.setResultsName('month') + _HYPHEN +
+      _TWO_DIGITS.setResultsName('day') +
+      text_parser.PyparsingConstants.TIME_MSEC_ELEMENTS +
+      # TODO: consider adding tz-offset support to parsers/text_parser.py?
+      pyparsing.Word(pyparsing.printables).setResultsName('time_zone_offset')
+  ).setResultsName('date_time')
+
+  # Multiline entry end marker, matched from right to left.
+  _GDS_ENTRY_END = pyparsing.StringEnd() | _GDS_DATE_TIME
```

Comment:
This seems to terminate early for some multi line log entries

Reply:
Try adding the class constant "BUFFER_SIZE = 16384"

Reply:
Thank you so much for the help debugging this! I re-ran it with the updated buffer size and received no errors, seems to be working as intended now.